doc/develop/getting_started/index.rst

```diff
@@ -487,6 +487,13 @@ Zephyr applications.
 It also contains additional host tools, such as custom QEMU and OpenOCD builds
 that are used to emulate, flash and debug Zephyr applications.
 
+.. note::
+
+    Although the Zephyr SDK is recommended for new users, alternative
+    toolchains are supported. Users may check the list of
+    :ref:`alternate toolchains <toolchains>` to see if an existing toolchain
+    they have installed is supported by Zephyr.
```

Comment:
I am not sure why this needs to be mentioned in the **Getting Started** Guide. Using a third-party toolchain is not really something for those who want to get started with Zephyr in the quickest and the most hassle-free way. Providing beginners with too many options at first may only confuse them and send them down the wrong path.

> to see if an existing toolchain they have installed is supported by Zephyr.

> they can likely use the GNU ARM embedded toolchain they already have installed and save on setup time and disk space

This will cause more problems for the people who are new to Zephyr because:

1. Third-party toolchains do not include any host tools, and people need to either install every host tool manually (not really something for the beginners) or download the Zephyr SDK anyway to get them.
2. There exist subtle yet very impactful differences between the Zephyr SDK and the third-party GNU toolchains (e.g. GNU Arm Embedded), which can lead to very confusing and hard to troubleshoot issues (see #43843; all third-party GNU toolchains should be deprecated eventually).
3. Third-party toolchains do not have the same level of support and userbase as the Zephyr SDK; so, you are likely to encounter more issues in the long term and any questions you ask about them will likely go unanswered. Moreover, if you have any problems using a **third-party GNU toolchain like the GNU Arm Embedded** (and you almost certainly will), your question will very likely be answered with "please try with the Zephyr SDK and let me know how it goes," in which case you might as well just install the Zephyr SDK in the first place to save the hassle.

For the reasons stated above, using a third-party toolchain is not something for the people who are getting started with Zephyr. IMHO, this should be dropped. Also note that using the instructions for using an alternate toolchain is already mentioned in the "Beyond Getting Started Guide," which is sufficient in my opinion.

Reply:
> Using a third-party toolchain is not really something for those who want to get started with Zephyr in the quickest and the most hassle-free way.

In my experience helping developers get started, it was. After reading through the discussions you mentioned, I see the reasoning behind potentially deprecating GNU ARM embedded in the future, but I think if that is a goal we have perhaps the solution here is not to add this note to the getting started guide, but to discuss *why* those developers struggled with Zephyr SDK (and what we could do to improve it)

From their feedback (as well as my personal experience), I gathered the following:
- The SDK is large- the disk space required to support toolchains for every architecture is significant. Maybe we should direct users to download the minimal SDK bundle instead, and provide guidance on choosing their architecture.
- The Windows SDK is built against python 3.8- since the chocolatey installer uses python 3.10 by default, SDK components like gdb that require python will fail to start until python 3.8 is installed. This is an issue I've seen on other systems as well (I recall running into it on Arch Linux), and maybe could be resolved by changing the chocolatey install instructions to use a specific python version.

Reply:
> to discuss why those developers struggled with Zephyr SDK (and what we could do to improve it)

Sure

> Maybe we should direct users to download the minimal SDK bundle instead, and provide guidance on choosing their architecture.

That sounds reasonable. I thought of doing that initially, but I was afraid the "provid[-ing] guidance on choosing their architecture" part could further confuse the people who are just getting started. If well stated enough, however, it could definitely improve developer experience.

> SDK components like gdb that require python will fail to start until python 3.8 is installed.

GNU Arm Embedded is no exception to this either. For this reason, we provide a separate GDB variant called "gdb-no-py" that builds without Python scripting support, and the Zephyr build system automatically defaults to using it if the required version of libpython is not locally available (by the way, from SDK 0.15.0 and on, the default "gdb" will become Python-less and a separate Python-capable variant called "gdb-py" will be provided).

> maybe could be resolved by changing the chocolatey install instructions to use a specific python version.

That sounds reasonable. The same can be done for macOS/Homebrew (`python@3.8`).

Reply:
> If we do this, I assume we would just update the required version of python when we updated getting started directions to use a new version of the SDK, correct?

Correct.

> I'm unsure how often the python dependency of GDB will change.

So far, it has not changed for almost 2 years; but, we will likely update to 3.10 soon since Ubuntu 22.04 has been released.

> (based on the change to using no-py in the next SDK, it sounds like this may not be an issue long term)

Sure, from the SDK 0.15.0 release and on, the default `gdb` executable (e.g. `arm-zephyr-eabi-gdb`) will no longer link against the libpython, and a separate Python-capable `gdb-py` (e.g. `arm-zephyr-eabi-gdb-py`) variant will be provided. This will make IDE integration, which may look for `*-gdb` by default, much easier.